Segfault during test suite

Not always... sometime
Will try to get a bactrace

```
OpenCensus Trace: Basic Class Method As Function Test [tests/basic_class_function.phpt]
OpenCensus Trace: Basic Function Test [tests/basic_function.phpt]
OpenCensus Trace: Basic Method Test [tests/basic_method.phpt]
OpenCensus Trace: Clear Traces [tests/clear_traces.phpt]
OpenCensus Trace: Nested spans [tests/nested_spans.phpt]
OpenCensus Trace: Basic Class Static Method [tests/static_method_test.phpt]
OpenCensus Trace: Trace Context [tests/trace_context.phpt]
```

All with `Termsig=11`

Some build warning also have to be fixed, as this can raised very strange issues

```
/dev/shm/BUILD/php-pecl-opencensus-0.0.1/ZTS/opencensus_trace.c:655:5: warning: implicit declaration of function 'opencensus_trace_span_minit' [-Wimplicit-function-declaration]
     opencensus_trace_span_minit(INIT_FUNC_ARGS_PASSTHRU);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/dev/shm/BUILD/php-pecl-opencensus-0.0.1/ZTS/opencensus_trace.c:656:5: warning: implicit declaration of function 'opencensus_trace_context_minit' [-Wimplicit-function-declaration]
     opencensus_trace_context_minit(INIT_FUNC_ARGS_PASSTHRU);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

--- a/Dockerfile
+++ b/Dockerfile
@@ -33,7 +33,7 @@ ENV TEST_PHP_ARGS="-q" \
     REPORT_EXIT_STATUS=1
 
 RUN phpize && \
-    ./configure --enable-stackdriver-trace && \
+    ./configure --enable-opencensus && \
     make clean && \
     make && \
     make test && \

--- a/cloudbuild.yaml
+++ b/cloudbuild.yaml
@@ -12,3 +12,11 @@ steps:
     args: ['build', '--build-arg', 'BASE_IMAGE=gcr.io/php-mvm-a/php72:alpha3', '.']
     waitFor: ['-']
     id: php72
+  - name: gcr.io/cloud-builders/docker
+    args: ['build', '--build-arg', 'BASE_IMAGE=gcr.io/php-stackdriver/php71-32bit', '.']
+    waitFor: ['-']
+    id: php71-32bit
+  - name: gcr.io/cloud-builders/docker
+    args: ['build', '--build-arg', 'BASE_IMAGE=gcr.io/php-stackdriver/php70-32bit', '.']
+    waitFor: ['-']
+    id: php70-32bit

--- a/ext/opencensus_trace.c
+++ b/ext/opencensus_trace.c
@@ -250,7 +250,7 @@ static opencensus_trace_span_t *opencensus_trace_begin(zend_string *function_nam
         php_mt_srand(GENERATE_SEED());
     }
 #endif
-    span->span_id = (php_mt_rand() >> 1);
+    span->span_id = ((uint32_t) php_mt_rand()) >> 1;
 
     if (OPENCENSUS_TRACE_G(current_span)) {
         span->parent = OPENCENSUS_TRACE_G(current_span);

--- a/ext/opencensus_trace.c
+++ b/ext/opencensus_trace.c
@@ -252,6 +252,12 @@ static opencensus_trace_span_t *opencensus_trace_begin(zend_string *function_nam
         php_mt_srand(GENERATE_SEED());
     }
 #endif
+    /**
+     * Force the random span id to be positive. php_mt_rand() generates 32 bits
+     * of randomness. On 32-bit systems, we must cast to an unsigned int before
+     * bitshifting to force a positive number. We're ok to lose on bit of
+     * randomness because previous versions of mt_rand only generated 31 bits.
+     */
     span->span_id = ((uint32_t) php_mt_rand()) >> 1;
 
     if (OPENCENSUS_TRACE_G(current_span)) {

--- a/ext/opencensus_trace.c
+++ b/ext/opencensus_trace.c
@@ -528,13 +528,13 @@ PHP_FUNCTION(opencensus_trace_function)
 {
     zend_string *function_name;
     zval *handler = NULL, *copy;
+    zval h;
 
     if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "S|z", &function_name, &handler) == FAILURE) {
         RETURN_FALSE;
     }
 
     if (handler == NULL) {
-        zval h;
         ZVAL_LONG(&h, 1);
         handler = &h;
     }
@@ -559,13 +559,13 @@ PHP_FUNCTION(opencensus_trace_method)
 {
     zend_string *class_name, *function_name, *key;
     zval *handler = NULL, *copy;
+    zval h;
 
     if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "SS|z", &class_name, &function_name, &handler) == FAILURE) {
         RETURN_FALSE;
     }
 
     if (handler == NULL) {
-        zval h;
         ZVAL_LONG(&h, 1);
         handler = &h;
     }

--- a/ext/opencensus_trace.c
+++ b/ext/opencensus_trace.c
@@ -16,6 +16,8 @@
 
 #include "php_opencensus.h"
 #include "opencensus_trace.h"
+#include "opencensus_trace_span.h"
+#include "opencensus_trace_context.h"
 #include "Zend/zend_compile.h"
 #include "Zend/zend_closures.h"
 #include "zend_extensions.h"

--- a/ext/opencensus_trace.c
+++ b/ext/opencensus_trace.c
@@ -519,7 +519,7 @@ void opencensus_trace_execute_internal(INTERNAL_FUNCTION_PARAMETERS)
 PHP_FUNCTION(opencensus_trace_function)
 {
     zend_string *function_name;
-    zval *handler, *copy;
+    zval *handler = NULL, *copy;
 
     if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "S|z", &function_name, &handler) == FAILURE) {
         RETURN_FALSE;
@@ -550,7 +550,7 @@ PHP_FUNCTION(opencensus_trace_function)
 PHP_FUNCTION(opencensus_trace_method)
 {
     zend_string *class_name, *function_name, *key;
-    zval *handler, *copy;
+    zval *handler = NULL, *copy;
 
     if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "SS|z", &class_name, &function_name, &handler) == FAILURE) {
         RETURN_FALSE;

--- a/ext/opencensus_trace_context.h
+++ b/ext/opencensus_trace_context.h
@@ -21,4 +21,6 @@
 
 extern zend_class_entry* opencensus_trace_context_ce;
 
+int opencensus_trace_context_minit(INIT_FUNC_ARGS);
+
 #endif /* PHP_OPENCENSUS_TRACE_CONTEXT_H */

--- a/ext/opencensus_trace_span.h
+++ b/ext/opencensus_trace_span.h
@@ -39,5 +39,6 @@ int opencensus_trace_span_add_label_str(opencensus_trace_span_t *span, char *k, 
 int opencensus_trace_span_apply_span_options(opencensus_trace_span_t *span, zval *span_options);
 opencensus_trace_span_t *opencensus_trace_span_alloc();
 void opencensus_trace_span_free(opencensus_trace_span_t *span);
+int opencensus_trace_span_minit(INIT_FUNC_ARGS);
 
 #endif /* PHP_OPENCENSUS_TRACE_SPAN_H */

